editoast/src/views/timetable/train_schedule.rs

```diff
@@ -590,10 +608,344 @@ async fn occupancy_blocks(
+            let (track_ranges, path_item_positions) = match pathfinding {
+                PathfindingResult::Success(pathfinding_result_success) => (
+                    &pathfinding_result_success.track_section_ranges,
+                    &pathfinding_result_success.path_item_positions,
+                ),
+                PathfindingResult::Failure(_) => {
+                    // The timetable can have a pathfinding failure for a train schedule
+                    // because the train schedule is not valid.
+                    // We skip it because we don't want to compute the track occupancy for an invalid train schedule.
+                    // It's not a problem because we are just looking for the track occupancy of the operational point for the valid trains.
+                    tracing::info!(train_schedule.id, "pathfinding failed");
+                    continue;
+                }
+            };
+
+            let path_projection = PathProjection::new(track_ranges);
+
+            // Get the positions and the times from the simulation
+            let report_train = match simulation {
+                Response::Success { final_output, .. } => final_output.report_train.clone(),
+                _ => {
+                    tracing::info!(train_schedule.id, "simulation failed");
+                    continue;
+                }
+            };
+
+            let path_item_id = match_path_item_id_with_op_id(
+                &path_item_cache,
+                &train_schedule,
+                &operational_point_id,
+            );
+
+            if let Some((track_section, track_occupancy)) = compute_track_occupancy(
+                path_item_id,
+                path_item_positions,
+                &path_projection,
+                &operational_point_track_offsets,
+                &report_train,
+                &train_schedule,
+            ) {
+                track_occupancy_map
+                    .entry(track_section)
+                    .or_default()
+                    .push(track_occupancy)
+            }
+        }
+        track_occupancy_map
+    };
+    Ok(Json(track_occupancy_map))
+}
+
+fn compute_track_occupancy_with_path_item_id(
+    train_schedule: &models::TrainSchedule,
+    path_item_id: &str,
+    path_item_positions: &[u64],
+    path_projection: &PathProjection,
+    operational_point_track_offsets: &[TrackOffset],
+    report_train: &ReportTrain,
+) -> Option<(String, TrackOccupancy)> {
+    let schedule_item = train_schedule
+        .schedule
+        .iter()
+        .find(|schedule| schedule.at.0 == path_item_id);
+    // Stop time at the operational point:
+    // Retrieve the stop time if it exists; otherwise, default to 0
+    let stop_duration = match schedule_item {
+        Some(schedule_item) => schedule_item.stop_for.clone().unwrap_or_default(),
+        // When the path item is the departure position without being a schedule point
+        None => PositiveDuration::default(),
```

Comment:
I'm not sure that I understand that case. Shouldn't we always find the `schedule_item` in the function `compute_track_occupancy_with_path_item_id` function? And if we do not find it, instead of defaulting to a `PositiveDuration::default()`, shouldn't we instead fallback to the other function `compute_track_occupancy_without_path_item_id`? If the answer to these questions are yes, then maybe we need see how to organize the code differently because something feels off.

Reply:
Got it. Thank you. Do you think we could make it more obvious for a future reader of that code (a comment? or something else?).